etl/src/main/java/org/pih/hivmigration/etl/sql/ExamLabResultsMigrator.groovy

```diff
@@ -0,0 +1,80 @@
+package org.pih.hivmigration.etl.sql
+
+class ExamLabResultsMigrator extends ObsMigrator {
+    @Override
+    def void migrate() {
+
+        executeMysql("Create hivmigration_exam_lab_results temp table", '''
+            CREATE TABLE hivmigration_exam_lab_results (
+                source_encounter_id INT,
+                lab_test varchar(80),
+                result varchar(255),
+                test_date date
+            );
+        ''')
+
+        loadFromOracleToMySql('''
+            INSERT INTO hivmigration_exam_lab_results (
+                source_encounter_id, lab_test, result, test_date
+            ) VALUES (?, ?, ?, ?)
+        ''',
+        '''
+            SELECT encounter_id, lab_test, result, test_date FROM HIV_EXAM_LAB_RESULTS
+        ''')
+
+        create_tmp_obs_table()
+
+        executeMysql("Migrate CD4", '''
+            INSERT INTO tmp_obs (source_encounter_id, concept_uuid, value_numeric)
+            SELECT
+                he.source_encounter_id,
+                concept_uuid_from_mapping('CIEL', '5497'),
+                CASE
+                    WHEN is_number(result) THEN TRIM(result)
+                    WHEN result REGEXP '^\\d*\\s*mm$' THEN extract_number(result)
+                    END
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN encounter e on he.encounter_id = e.encounter_id
+            WHERE
+                lab_test = 'cd4'    
+                AND is_number(result) OR result REGEXP '^\\d*\\s*mm$';
```

Comment:
Would be nice to give an example that matches that REGEXP, since regexs are so hard to read.

Reply:
Good call!